GA_core/ga_classes.py

```diff
@@ -5,26 +5,51 @@ def update(self, generation, best_score):
 
 
 class Genetic_Algorithm:
+    """
+    gene = policy (per SKU)
+
+    chromosome = individual solution (e.g., for 100 SKUs, max 100 policies)
+
+    population = a number of chromosomes
+    """
 
     def __init__(self) -> None:
         self.observers = []
         
         def create_population(self) -> None:
+            """
+            create some number of Individual_Solution objects
+            """
             pass
         
         def evaluate_population(self) -> None:
+            """
+            find the fitness of each chromosome (i.e., the cost function)
+            """
             pass
 
         def select_parents(self) -> None:
+            """
+            select parent chromosomes for a crossover

```

Comment:
Correct. In most cases you use the fittest to do so. However in practise you need some randomness. There is a of theory on this field check the **roulette wheel selection**